### DISCUSSION TITLE
Buffer error on Android mobile

### DISCUSSION THREAD

**Discussion by Arminius4** [Vinzent03/obsidian-git#350] [Problems]
Is anyone else also getting this error when trying to use mobile app on Android?
I had / have this on two devices already (it miraculously disappeared on one device).

It happens when trying to perform git-actions using the source control view buttons. My problem is, that I do not know what the issue could be and that the error message ("Buffer error") is so generic. 

On my second device, everything runs fine now, even though I have not done anything...

**Comment by PEConn** [comment]
I'm getting something similar - everything works fine on my phone, but I can't push or pull on my tablet because of it. Is there any way to see the console logs on Android?

**Comment by PEConn** [comment]
I found Issue #307 which suggests that you can go into Android Settings and give Obsidian Unrestricted battery use, unfortunately that didn't work.

If I attach the device to `adb logcat`, when I try to trigger a backup, I get:

```
19685-19910/md.obsidian E/BLASTBufferQueue: [VRI[MainActivity]#3](f:0,a:2) Faking releaseBufferCallback from transactionCompleteCallback
```

repeated about 20 times.

**Reply by kreibaum** [reply]
I am also getting the "buffer error" when doing git push. Pulling is fine though.
I don't have adb set up to figure out if I get the same kind of error messages though.

Has been working fine before, I wonder if this is because I edited something on the desktop version and pushed there. I didn't have to handle and conflicts on mobile before, only had to merge once on desktop.


Update: For me the issue went away after I waited three hours and tried again.

**Comment by t-schreibs** [comment]
I keep running into this from time to time as well, can't seem to figure out what might be causing it. Restarting my phone seems to fix it though?

**Reply by bmrussell** [reply]
Same here, no need for a full phone reboot, I just worked around it by force closing Obsidian.

**Comment by RuVl** [comment]
I'm getting buffer error when doing git pull/push.
For me the issue went away after I update all plugins.

**Comment by siothelibrarian** [comment]
I'm having this same issue, running Androind on my Pixel 6a. I set up the repository yesterday and it was working fine. Today I have the "buffer error" and am unable to push changes from my phone. I've force closed obsidian, restarted the device, and updated plugins and that hasn't done anything... any other ideas?

**Comment by blockspacer** [comment]
I'm having this same issue, running Android 13 on galaxy tab s7

**Comment by jmessi** [comment]
Having this same issue for the past 2-3 weeks. It pulls everything correctly, but can't push any changes made on the device

**Comment by kaydenkehe** [comment]
Same issue on a Galaxy S10e!

**Comment by Gander7** [comment]
Same Issue for Samsung Galaxy S20G.
I force restarted app then restarted phone and it worked. Took a few minutes to figure itself out after the restart.

**Reply by Gander7** [reply]
This happened to me after I added a bunch of markdown files from my computer, then when I opened my mobile device it added a commit to delete all the files I just added. So I've spent the morning adding them back from the commit history which is when it triggered "buffer overflow".
I had `.obsidian/workspace*` files tracked but I've added them to `.gitignore` which looking at the commit, having them tracked may have been a factor.

**Comment by Meddrin** [comment]
For me, closing the app, waiting 1-2 minutes and restarting solves the issue.

**Comment by 7Codez** [comment]
Who among you knows a real reason for the buffer problem that I suffer from every now and then.  

And what is the final solution?

**Comment by gfordinal** [comment]
buffer error still happening. I the morning it worked fine, a few files added and it gives me that error on pulling and on committing.

**Comment by igorsblv** [comment]
I'm having this same issue, running Androind on Xiaomi. I set up the repository yesterday and it was working fine. Today I have the "buffer error" and am unable to push changes from my phone. I've force closed obsidian, restarted the device, and updated plugins and that hasn't done anything...

**Comment by tyrantqiao** [comment]
device: xiaomi mix4

resolution don't know why but work

- gitignore remove all .obsidian/* 
- git plugin disable all auto backup config, only remain pull
- stage file---commit stage --- push

**Reply by siothelibrarian** [reply]
I tried this, and it worked for less than 24 hours before giving me a new issue -  when I try to stage, commit, or push, I seem to get no response from the app other than it asking me to type in a commit message. Then there's a "Getting Status: This takes a while" message and... nothing. I've pushed the files multiple times from the app with no actual change. I have another app that I use to sync manually, and I can see all the changes - they haven't been committed or pushed.

**Comment by anibaldeboni** [comment]
Same problem on iOS 17.1.1. Can't push anything

**Comment by UnusualFrog** [comment]
I'm also getting this problem on my Galaxy s20
Running android 13

**Comment by MartinVoplakal** [comment]
The same problem. That's a shame that I can not use this plugin on mobile because of that. Hopefully I see an effort to solve this issue and to take this plubin on different level by replacing git integrating to WebAssembly alternativ. https://github.com/denolehov/obsidian-git/issues/572

**Comment by siothelibrarian** [comment]
I gave up and switched to Syncthing. It's free, it's open source, it works
on a variety of platforms, it took me less than an hour to set up, and I've
had zero issues with conflicts in the month I started using it. Much better
option than this plugin that mysteriously doesn't work and a dev that
offers no assistance.

**Comment by mahe4** [comment]
Same problem here.
too sad, that this isn't even addressed for years on end.

**Reply by sirEven** [reply]
Resurrecting with a technical question / request: Would it be possible to implement a settings option, where the user can increase the buffer size for push and pull on mobile?
I ran into a buffer error on my mac, when pushing far too many huge files, which was fixable by setting buffer size to 500mb (`git config --global http.postBuffer 524288000`).
Right now my obsidian on mobile is crashing on start up as it tries to fetch that huge amount, and runs into a buffer error as well. This is happening on iOS.

**Reply by gfordinal** [reply]
I guess the problem here is, that this is not "real" git, but rather a un-performant reimplementation in Javascript.
Which occurs peculiar to me, since Android has a Unix Kernel and should just run any git implementation just fine.

Though since it doesn't, maybe there are no global git configs to define. I hope, someone corrects me on that part!
